### PR TITLE
docs(example): show folks how to use pull-streams instead

### PR DESCRIPTION
**Pull request by dignifiedquire**
Always stream incoming data through pull streams and use browserify-aes directly from master now. I was able to add a 300Mb sized file with this quite quickly.

Fixes most issues in https://github.com/ipfs/js-ipfs/issues/952  I believe

### CODE REVIEW

File: examples/exchange-files-in-browser/public/index.html [hunk 1]
```diff
@@ -62,7 +62,7 @@
     </div>
 
     <!-- The IPFS node module -->
-    <script src="//unpkg.com/ipfs/dist/index.min.js"></script>
+    <!-- <script src="//unpkg.com/ipfs/dist/index.min.js"></script> -->
```
Comment (diasdavid) [inline examples/exchange-files-in-browser/public/index.html]: This was supposed to be the example that showed that "no extra batteries required". Kind of sad we lost that.

File: examples/exchange-files-in-browser/public/js/app.js [hunk 2]
```diff
@@ -119,93 +120,44 @@ function onDrop (event) {
...
-          })
+  pull(
+    pull.values(files),
+    pull.through((file) => console.log('Adding %s', file)),
+    pull.asyncMap((file, cb) => pull(
+      pull.values([{
+        path: file.name,
+        content: pullFilereader(file)
+      }]),
+      node.files.createAddPullStream(),
```
Comment (diasdavid) [inline examples/exchange-files-in-browser/public/js/app.js]: This is awesome, but we shouldn't have examples with undocumented (and already agreed to change) API

Needs this solved first. https://github.com/ipfs/interface-ipfs-core/issues/126

**Comment by diasdavid** [comment]
Awesome job @dignifiedquire! I'll keep this PR open until https://github.com/ipfs/js-ipfs/pull/988#discussion_r136845093 is solved

File: examples/exchange-files-in-browser/public/index.html (hunk 1 above)
Comment (VictorBjelkholm) [reply examples/exchange-files-in-browser/public/index.html]: @diasdavid unpkg doesn't allow linking to a branch? We could also publish latest master ourselves on IPFS and link that instead

**Comment by VictorBjelkholm** [comment]
Hm, a bit worried about this as most people haven't been exposed to pull-streams and we're supposed to support the API that most people use as well. This seems to amount to that even though you can use js-ipfs without using pull-streams, it's really inefficient and you should probably use pull-streams.

As this is supposed to be a simple example, I'm wondering if it's not best to stick with the non-pullstreams version.

**Comment by VictorBjelkholm** [comment]
Maybe we could show it's done with non-pullstreams in comments (risk of having code that is not being run though, might not actually work...) or make the pullstream into a separate example for better performance.

File: examples/exchange-files-in-browser/public/index.html (hunk 1 above)
Comment (dignifiedquire) [reply examples/exchange-files-in-browser/public/index.html]: We don't have to keep this, with the new release that would be fine to use the unkpkg version again

**Comment by dignifiedquire** [comment]
We are currently alienating people with the browser crashing, so I think having pull-streams is much better than crashing and being slow. node streams are really not meant to be used in the browser in most situations I don't think it's good to show examples with them, rather use non streaming methods by default and educate users about better options

**Comment by diasdavid** [comment]
We will have both API and once they are properly documented, we can use them in the examples. This is one of the priorities for 0.27 (and probably the target thing to get it released)

**Comment by diasdavid** [comment]
Actively working on https://github.com/ipfs/interface-ipfs-core/pull/162 this week to then finish up this PR

**Comment by diasdavid** [comment]
Who was interested on this PR, please review https://github.com/ipfs/interface-ipfs-core/pull/162#issuecomment-341691928

**Comment by diasdavid** [comment]
Redirecting the efforts here to this PR https://github.com/ipfs/js-ipfs/pull/1086

**Comment by mitra42** [comment]
@diasdavid - it would be useful to leave issues open until they are solved, even if the solution is elsewhere.  I for one have been watching this thread in the hope of finding the docs implied by its title, " show folks how to use pull-streams instead  "  with the direction to #1086 , I can't find any answer to this question in that issue nor in ipfs/interface-ipfs-core#162, and I'm probably not the only one waiting on docs before converting app code to use pull-streams.

**Comment by diasdavid** [comment]
@mitra42 valid point. I tend to keep issues and PR open until they are fixed or converge elsewhere. It is hard to keep track of multiple issues discussing the same things and that is why I always leave a pointer saying that "x discussion continues at issue y".

@mitra42, for your case, https://github.com/ipfs/interface-ipfs-core/pull/162 brought you something even better as you won't have to worry about pull-streams and just use ipfs.files.add with files and get hashes back. See the updated https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/FILES.md#add

See the updated example on https://github.com/ipfs/js-ipfs/pull/1086 that this PR was also updating (direct link here https://github.com/ipfs/js-ipfs/blob/f5e756106f5f3a4cd92cbea43863b35bf8d90a99/examples/exchange-files-in-browser/public/js/app.js#L102-L135),  as you will see there are no Readable or Pull Streams involved. I used this code files bigger than 750Mb, a significant improvement.

**Comment by mitra42** [comment]
Email sent as this goes into multiple topics.

**Pull request closed without merging**